Deploy and Manage: Elastic Cloud Enterprise Deploy (#499)
elastic/docs-content: deploy-manage/deploy/cloud-enterprise.md

---
applies:
  ece: all
---

# Elastic Cloud Enterprise [Elastic-Cloud-Enterprise-overview]

> {{ece}} (ECE) is an Elastic self-managed solution for deploying, orchestrating, and managing {{es}} clusters at scale. It provides a centralized platform that allows organizations to run {{es}}, {{kib}}, and other {{stack}} components across multiple machines.

Reviewer: this page is missing the key info from the ECE overview below (the "why ECE")

Refer to [](./cloud-enterprise/ece-overview.md) for a detailed introduction to ECE, including its features, use cases, and architecture.

::::{tip}
If you are looking for a solution to orchestrate and manage {{es}} clusters natively on Kubernetes, consider using [Elastic Cloud on Kubernetes (ECK)](./cloud-on-k8s.md) instead of ECE. ECK enables you to orchestrate Elastic Stack applications seamlessly on Kubernetes, leveraging it as the underlying platform for deployment, scaling, and lifecycle management.
::::

> % should we use a L2 heading here or just continue?

Reviewer: I like the L2

## Section overview

> This section focuses on deploying ECE and the orchestrating and configuring {{es}} clusters, also referred to as `deployments`.

Reviewer: ```suggestion
This section focuses on deploying ECE and orchestrating and configuring {{es}} clusters, also referred to as `deployments`.
```

In ECE, a deployment is a managed {{stack}} environment that provides users with an {{es}} cluster along with supporting components such as {{kib}} and other optional services like APM and Fleet.

This section covers the following tasks:

> * [Deploy ECE](./cloud-enterprise/deploy-an-orchestrator.md)

Reviewer: might want to specify that this is the orchestrator / installation - this always confused me in the original doc

Author: Done it

    - [Prepare the environment](./cloud-enterprise/prepare-environment.md)
    - [Install ECE](./cloud-enterprise/install.md)
    - [Air gapped installations](./cloud-enterprise/air-gapped-install.md)
    - [Configure ECE](./cloud-enterprise/configure.md)

* [Work with deployments](./cloud-enterprise/working-with-deployments.md)
  - Use [](./cloud-enterprise/deployment-templates.md) to [](./cloud-enterprise/create-deployment.md)
  - [](./cloud-enterprise/customize-deployment.md)
  - Use the deployment [Cloud ID](./cloud-enterprise/find-cloud-id.md) and [Endpoint URLs](./cloud-enterprise/find-endpoint-url.md) for clients connection

> * Learn how to use the multiple [](./cloud-enterprise/tools-apis.md) available

Reviewer: this page won't be instructional

```suggestion
* Learn about [](./cloud-enterprise/tools-apis.md) that you can use with ECE
```

> Other sections of the documentation also include important tasks related with ECE:

Reviewer 2: ```suggestion
Other sections of the documentation also include important tasks related to ECE:
```

> * [Secure your ECE installation](../security/secure-your-elastic-cloud-enterprise-installation.md)

Reviewer: technically, the rest of the security section might be relevant as well (e.g. securing settings, saved objects?) 

could call out the ece section specifically though

Author: Secure your ECE installation was a page in the original content with a lot of security related topics for ECE as chid docs. I hope to have something similar in the security section there.

It's true that I expect that to be focused on the orchestrator security and not on deployments security.

I'll add a link to the "secure your deployments" area which was  https://www.elastic.co/guide/en/cloud-enterprise/current/ece-securing-clusters.html in the original docs. Thanks!

What I wouldn't do linking in the main page of ECE to something as specific as `kibana saved objects`, but if you share other examples that I might be missing please share!

For `secure settings` I expect them to be part of the security content mentioned previously.

* [Users and roles](../users-roles/cloud-enterprise-orchestrator.md)
* [Manage snapshot repositories](../tools/snapshot-and-restore.md)
* [Manage licenses](../license/manage-your-license-in-ece.md)
* [ECE platform maintenance operations](../maintenance/ece.md)

## How ECE differs from Elastic Cloud and other orchestrators

For information about other deployment options, refer to [](../deploy.md).

% - **Elastic Cloud** (hosted by Elastic) is fully managed and runs on AWS, GCP, and Azure.
% - **ECE** is for organizations that **want to self-host and manage their own Elastic Stack** deployments across their infrastructure.
% - **ECK** is for organizations that want to run on Kubernetes.

## Supported versions [ece-supported-versions]

> Refer to [Elastic Support Matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise) for more information about supported Operating Systems, Docker, and Podman versions.

Reviewer: ```suggestion
Refer to the [Elastic Support Matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise) for more information about supported Operating Systems, Docker, and Podman versions.
```

* Host your regulated or sensitive data on your internal network.
* Reuse your existing investment in on-premise infrastructure and reduce total cost.
* Maximize the hardware utilization for the various clusters.
* Centralize the management of multiple Elastic deployments across teams or geographies.

Refer to [](./cloud-enterprise/ece-architecture.md) for details about the ECE platform architecture and the technologies used.

## ECE features

- **Automated scaling & orchestration**: Handles cluster provisioning, scaling, and upgrades automatically.
- **High availability & resilience**: Ensures uptime through multiple Availability Zones, data replication, and automated restore and snapshot.
- **Centralized monitoring & logging**: Provides insights into cluster performance, resource usage, and logs.
- **Single Sign-On (SSO) & role-based access aontrol (RBAC)**: Allows organizations to manage access and security policies.
- **API & UI management**: Offers a web interface and API to create and manage clusters easily.
- **Air-gapped installations**: Support for off-line installations.
- **Microservices architecture**: All services are containerized through Docker.

> Check the [glossary](https://www.elastic.co/guide/en/elastic-stack-glossary/current/terms.html) to get familiar with the terminology for ECE as well as other Elastic products and solutions.

Reviewer: ```suggestion
Check the [glossary](asciidocalypse:///docs-content/docs/reference/glossary.md) to get familiar with the terminology for ECE as well as other Elastic products and solutions.
```
